File: 이상욱/BOJ/m2w3/BOJ20115.java
package m2w8;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collections;
import java.util.StringTokenizer;

public class BOJ20115 {
	
	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		int n = Integer.parseInt(br.readLine());
		
		Double[] arr = new Double[n];
		
		st = new StringTokenizer(br.readLine());
		
		for(int i = 0; i < n; i++) {
			arr[i] = Double.parseDouble(st.nextToken());
		}
		
		Arrays.sort(arr,Collections.reverseOrder());
		
		double sum = arr[0];
		
		for(int i = 1; i < n; i++) {
			if(sum+arr[i]/2 > sum/2 + arr[i]) {
				sum = sum + arr[i]/2;
			}else {
				sum = sum/2 + arr[i];
			}
		}
		
		System.out.println(sum);
		
	}
}

File: 이상욱/BOJ/m2w8/BOJ20115.kt
package m2w8

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main(){

    val br = BufferedReader(InputStreamReader(System.`in`))

    val n = br.readLine().toInt()
    val arr = DoubleArray(n)

    val st = StringTokenizer(br.readLine())

    for (i in 0 until n){
        arr[i] = st.nextToken().toDouble()
    }

    arr.sortDescending()
    var sum : Double = arr[0]

    for (i in 1 until n){
        sum = if(sum + arr[i]/2 > sum/2 + arr[i]){
            sum+arr[i]/2
        }else{
            sum/2 + arr[i]
        }
    }

    println(sum)
}


